### ISSUE TITLE
New rule to enforce array parameters format

### ISSUE THREAD

**Issue by tatomyr** [Redocly/redocly-cli#1317]
**Is your feature request related to a problem? Please describe.**

There's no built-in rule to enforce explicit descriptions of array-type parameters (in particular, having `style` and `explode` properties set). It's possible to define a configurable rule to require those fields for all types of parameters, but they don't really make sense for non-array ones. 

**Describe the solution you'd like**

A rule similar to [boolean-parameter-prfixes](https://redocly.com/docs/cli/rules/boolean-parameter-prefixes/) that enforces explicit description of  parameters of array type, so properties `style` and `explode` are required:

```yaml
components:
  parameters:
    ArrayParameter:
       in: query
       style: form      # <-- required
       explode: true  # <-- required
       schema:
           type: array # <-- either explicitly defined as an array
           items: ...     # <-- or implicitly by having the `items` field

       
```

Naming to be discussed, maybe `explicit-parameter-format`? 

It also could have some configuration, like the following:

```yaml
rules:
  explicit-parameter-format: 
     severity: error
     in: 
        - header
        - query
```

**Describe alternatives you've considered**

Writing a custom rule or requiring `style` and `explode` fields to be set for all types of parameters via a configurable rule -- each option has its drawbacks.

**Additional context**

Related discussion: https://redoc-ly.slack.com/archives/C02UHE7EL3E/p1698886138140729

**Comment by RomanHotsiy** [comment]
I raised it to p2: the impact is minimal but the confidence is high and the implementation is very easy.

**Comment by RomanHotsiy** [comment]
Raising to p1. It's getting important for one of our customers.

**Comment by lornajane** [comment]
Rule naming: how about `array-parameter-fields` since it requires additional fields on the array params, and the rule that applies only to boolean params is named `boolean-parameter....`

**Comment by RomanHotsiy** [comment]
I like the `array-parameter-` prefix. How about `array-parameter-serialization`?

**Comment by lornajane** [comment]
The `style` parameter is about serialization but I'm not sure that `explode` is for that purpose, or perhaps not only for that purpose - and so I ended up with the more vague `fields` suffix. The relevant specification section: https://spec.openapis.org/oas/v3.1.0#parameter-object

I don't feel strongly either way about using `serialization`

**Comment by RomanHotsiy** [comment]
I believe `explode` is about serialization too. Check this out: https://swagger.io/docs/specification/serialization/

**Comment by tatomyr** [comment]
Resolved in v1.5.0.
